misc/userscripts/qute-pass

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+# Insert login information using pass and a dmenu-provider (e.g. dmenu, rofi -dmenu, ...).
+# A short demonstration can be seen here: https://i.imgur.com/KN3XuZP.gif.
+#
+# The domain of the site has to appear as a segment in the pass path, for example: "github.com/cryzed" or
+# "websites/github.com". How the username and password are determined is freely configurable using the CLI arguments.
+# The login information is inserted by emulating key events using qutebrowser's fake-key command in this manner:
+# [USERNAME]<Tab>[PASSWORD], which is compatible with almost all login forms.
+#
+# Dependencies: tldextract (Python 3 module), pass
+# For issues and feedback please use: https://github.com/cryzed/qutebrowser-userscripts
+#
+# Chris Braun (cryzed), 2017
+
+import argparse
+import enum
+import fnmatch
+import functools
+import os
+import re
+import shlex
+import subprocess
+import sys
+
+import tldextract
+
+PASSWORD_STORE_PATH = os.path.expanduser('~/.password-store')
+DMENU_INVOCATION = 'rofi -dmenu'
+ENCODING = 'UTF-8'
+
+argument_parser = argparse.ArgumentParser()
+argument_parser.add_argument('url', nargs='?', default=os.getenv('QUTE_URL'))
```

Comment:
Is there a reason you use `os.getenv` over `os.environ['QUTE_URL']` here?

Reply:
Yes, initially for debugging. I wanted to test various URLs/domains on the command line, outside of the qutebrowser environment -- accessing `os.environ` then directly would throw an exception because of the missing key, where as getenv returns the default `None` value. Also not needed anymore.